File: April-2023/Java/RemovingStarsFromAString.java
// 2390. Removing Stars From a String
class Solution {
    public String removeStars(String s) {
        Stack<Character> stack = new Stack<>();
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i)!='*'){
                stack.push(s.charAt(i));
            }
            else{
                stack.pop();
            }
        }
        StringBuilder sb = new StringBuilder();     
        while(!stack.isEmpty()){
            sb.append(stack.pop());         
        }   
        return sb.reverse().toString();
    }
}

File: April-2023/C++/RemovingStarsFromAString.cxx
// 2390. Removing Stars From a String
class Solution {
public:
    string removeStars(string s) {
    stack<char> stack;
    for (int i = 0; i < s.length(); i++) {
        if (s[i] != '*') {
            stack.push(s[i]);
        } else {
            stack.pop();
        }
    }
    string result = "";
    while (!stack.empty()) {
        result += stack.top();
        stack.pop();
    }
    reverse(result.begin(), result.end());
    return result;
}
};
